### DISCUSSION TITLE
Access initState() and dispose() in a ConsumerWidget

### DISCUSSION THREAD

**Discussion by julienlebren** [rrousselGit/riverpod#557] [Q&A] [answered]
👋

Everything is in the title. How can I achieve that?

I want to init some stuff when the widget is created, I was using a `StatefulWidget` until Riverpod 0.14 but I'm migrating to 1.0 and since `context.read` is no longer available, I had to switch to `ConsumerWidget` to get the `WidgetRef` in my build method.

But my `initState` and `dispose` methods are no longer available. Did I missed something?

**Comment by TimWhiting** [comment, marked as answer]
There is a ConsumerStatefulWidget that has a `ref` property that you can access in `initState` / `dispose`
